spec/index.bs

```diff
@@ -289,15 +291,34 @@ The {{Navigator/gpu}} attribute is used to access a {{GPU}} object from the main
 ## GPU ## {#gpu-interface}
 
 <script type=idl>
-[Exposed=Window]
+[Exposed=Window,DedicatedWorker]
 interface GPU {
-    // May reject with DOMException  // TODO: DOMException("OperationError")?
-    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options = {});
+    Promise<sequence<GPUAdapter>> requestAdapters(optional GPURequestAdapterOptions options = {});
```

Comment:
nit: I kinda like the idea you surfaced internally of having a `Promise<GPUAdapterList>` that's on `GPU` and contains the `gpuadapterschanged` event and that you could query synchronously for adapters. It makes sense: the system or GPU process can continuously push updates to this object so only its initial creation needs to be asynchronous.

Reply:
I decided against this, but given https://github.com/gpuweb/gpuweb/pull/522#discussion_r359470045 I will look into it again.

Reply:
Ok, so the problem with GPUAdapterList is there's no way to signal to the application (e.g. in a background tab): "I can't give you an adapter yet, but I probably will, so don't fallback yet".

Haven't yet come up with a better way to do that, but still working through it.

Reply:
alternatives:
- `gpu { adapterList { getAdapters(), onadapteradded } }`. adapterList can get 'revoked' and replaced with a new, pending promise to indicate this state. Problems: no signal for when adapterList goes from rejected (app should use fallback) to pending (app should consider initializing webgpu), because onadapteradded is basically deregistered on adapterList replacement.
- `gpu { onadapteradded, adapterList { getAdapters() } }`. adapterList can get 'revoked'. onadapteradded indicates a signal to await and probe the adapterList (regardless of whether the app is already using webgpu or whether the adapterList got replaced). Problem: basically the same as `gpu { requestAdapters() }` but more complex